extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java

```diff
@@ -115,15 +118,14 @@ public DataSegment push(File inDir, DataSegment segment) throws IOException
           storageDir,
           segment.getShardSpec().getPartitionNum()
```

Comment:
`outIndexFile` could be created using `JobHelper.makeFileNamePath()`. And to do that, we can pull out the `makeFileNamePath()` method from `JobHelper` to `DataSegmentPusher`. Doing that will ensure no separate logic for index file path creation at 2 places.  
We can do the same for tmp file and `descriptor.json`.